.github/scripts/lint/lint.py

```diff
@@ -0,0 +1,177 @@
+#!/usr/bin/python3
+
+"""
+Custom linter for chisel slice definition files.
+
+Use in addition with yamllint (https://yamllint.readthedocs.io), since this
+script only covers slice definition file specifc rules.
+"""
+
+import argparse
+import sys
+import typing
+from dataclasses import dataclass
+
+import yaml
+
+
+def parse_args() -> argparse.Namespace:
+    """
+    Parse CLI args passed to this script.
+    """
+    parser = argparse.ArgumentParser(
+        description="Lint slice definition files",
+    )
+    parser.add_argument(
+        "files",
+        metavar="file",
+        help="Chisel slice definition file(s)",
+        nargs="*",
+    )
+    parser.add_argument(
+        "--sorted-slices",
+        action=argparse.BooleanOptionalAction,
+        help="Slice names must be sorted",
+    )
+    parser.add_argument(
+        "--sorted-essential",
+        action=argparse.BooleanOptionalAction,
+        help="Entries in 'essential' must be sorted",
+    )
+    parser.add_argument(
+        "--sorted-contents",
+        action=argparse.BooleanOptionalAction,
+        help="Entries in 'contents' need to be sorted",
+    )
+    return parser.parse_args()
+
+
+def is_sorted(entries: list) -> bool:
+    """
+    Return true if a list is sorted in ASCENDING order.
+    """
+    for i in range(len(entries) - 1):
```

Comment:
A simpler way of writing this function would be just: 
```
def is_sorted(entries: list) -> bool:
    return sorted(entries) == entries
```
Surprisingly, `sorted` function is more optimized because it is implemented in C, and would usually give bteer benchmark results than a python for loop. However, in this particular case, the difference is very much negligible.

Reply:
Added in https://github.com/canonical/chisel-releases/pull/146/commits/412012b9d0f30c319fef7813f5e99ca99c9265d8, thanks!